action.yml

```diff
@@ -28,19 +28,53 @@ runs:
       with:
         python-version: '3.9'
 
-    - uses: hashicorp/setup-terraform@v2
-      with:
-        terraform_version: ${{ inputs.terraform_version }}
-        terraform_wrapper: false
+    - name: Setup tfenv
+      run: |
+        if [[ ! -d ~/.tfenv ]]; then
+          git clone -b v3.0.0 https://github.com/tfutils/tfenv.git ~/.tfenv
+        fi
+        sudo ln -sf ~/.tfenv/bin/* /usr/local/bin
+      shell: bash
 
     - run: pip install jinja2
       shell: bash
 
     - id: check
       run: |
         set +e
-        python3 ${{ github.action_path }}/tfcheck.py ${{ inputs.directory }} -report report.md | tee >(tail -1 > result.txt)
-        echo "returncode=${PIPESTATUS[0]}" >> $GITHUB_OUTPUT
+        read -r -a directories <<< "${{ inputs.directory }}"
+        read -r -a versions <<< "${{ inputs.terraform_version }}"
+
+        len=${#directories[@]}
+
+        returncode=0
+        for (( i=0; i<len; i++)); do
+          # Append three dashes between reports
+          if [ "$i" -gt 0 ]; then
+            printf "\n\n---\n\n" >> report.md
+          fi
+
+          # Change Terraform version
+          tfenv install "${versions[$i]}"
```

Comment:
Would this work if only one version is given?

Reply:
Yeah, let me show you on the other repo

Reply:
I see. Seems like shell can handle out-of-range list, and tfenv commands can handle unspecified version. Cool!